File: src/main/scala/kim/dencka/scala/easy/MissingNumber.scala
package kim.dencka.scala.easy

object MissingNumber extends App {

  def missingNumber(nums: Array[Int]): Int =
    nums.sorted.fold(0) { case (cnt, el) =>
      if (cnt == el) cnt + 1
      else cnt
    }

  println(missingNumber(Array(0, 1, 2, 3, 4, 5, 6, 7, 8, 9)))

}

File: src/main/scala/kim/dencka/java/easy/MissingNumber.java
package kim.dencka.java.easy;

class MissingNumber {
    public static void main(String[] args) {
        int[] nums = {3, 0, 1};
        int[] nums2 = {9, 6, 4, 2, 3, 5, 7, 0, 1};
        System.out.println(missingNumber(nums));
        System.out.println(missingNumber(nums2));
    }

    public static int missingNumber(int[] nums) {
        int k = nums.length;
        int full = k * (k + 1) / 2;
        int sum = 0;
        for (int num : nums) {
            sum += num;
        }
        return full - sum;
    }
}
